addons/keda/keda.go

```diff
@@ -0,0 +1,551 @@
+					APIVersion: kedav1alpha1.SchemeGroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: e.Integration.Namespace,
+					Name:      extConfigName,
+				},
+			}
+			for _, k := range util.SortedStringMapKeys(trigger.authentication) {
+				triggerAuth.Spec.SecretTargetRef = append(triggerAuth.Spec.SecretTargetRef, kedav1alpha1.AuthSecretTargetRef{
+					Parameter: k,
+					Name:      extConfigName,
+					Key:       k,
+				})
+			}
+			e.Resources.Add(&triggerAuth)
+			authenticationRef = &kedav1alpha1.ScaledObjectAuthRef{
+				Name: extConfigName,
+			}
+		} else if trigger.AuthenticationSecret != "" {
+			s := v1.Secret{}
+			key := ctrl.ObjectKey{
+				Namespace: e.Integration.Namespace,
+				Name:      trigger.AuthenticationSecret,
+			}
+			if err := e.Client.Get(e.Ctx, key, &s); err != nil {
+				return errors.Wrapf(err, "could not load secret named %q in namespace %q", trigger.AuthenticationSecret, e.Integration.Namespace)
+			}
+			// Fill a TriggerAuthentication from the secret
+			triggerAuth := kedav1alpha1.TriggerAuthentication{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "TriggerAuthentication",
+					APIVersion: kedav1alpha1.SchemeGroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: e.Integration.Namespace,
+					Name:      extConfigName,
+				},
+			}
+			sortedKeys := make([]string, 0, len(s.Data))
+			for k := range s.Data {
+				sortedKeys = append(sortedKeys, k)
+			}
+			sort.Strings(sortedKeys)
+			for _, k := range sortedKeys {
+				triggerAuth.Spec.SecretTargetRef = append(triggerAuth.Spec.SecretTargetRef, kedav1alpha1.AuthSecretTargetRef{
+					Parameter: k,
+					Name:      s.Name,
+					Key:       k,
+				})
+			}
+			e.Resources.Add(&triggerAuth)
+			authenticationRef = &kedav1alpha1.ScaledObjectAuthRef{
+				Name: extConfigName,
+			}
+		}
+
+		st := kedav1alpha1.ScaleTriggers{
+			Type:              trigger.Type,
+			Metadata:          meta,
+			AuthenticationRef: authenticationRef,
+		}
+		obj.Spec.Triggers = append(obj.Spec.Triggers, st)
+	}
+	e.Resources.Add(&obj)
+	return nil
+}
+
+func (t *kedaTrait) hackControllerReplicas(e *trait.Environment) error {
+	ctrlRef := t.getTopControllerReference(e)
+	if ctrlRef.Kind == camelv1alpha1.KameletBindingKind {
+		// Update the KameletBinding directly (do not add it to env resources, it's the integration parent)
+		key := ctrl.ObjectKey{
+			Namespace: e.Integration.Namespace,
+			Name:      ctrlRef.Name,
+		}
+		klb := camelv1alpha1.KameletBinding{}
+		if err := e.Client.Get(e.Ctx, key, &klb); err != nil {
+			return err
+		}
+		if klb.Spec.Replicas == nil {
+			one := int32(1)
+			klb.Spec.Replicas = &one
+			if err := e.Client.Update(e.Ctx, &klb); err != nil {
+				return err
+			}
+		}
+	} else if e.Integration.Spec.Replicas == nil {
+		one := int32(1)
+		e.Integration.Spec.Replicas = &one
+		// Update the Integration directly as the spec section is not merged by default
```

Comment:
nit: Maybe patching the `scale` endpoint could be used instead, rather than updating the whole resource:

https://github.com/apache/camel-k/blob/fef775de02b3a59b9c844ba8bf9c418bf8a708b7/e2e/common/scale_integration_test.go#L79